### PR TITLE
38 coverimage management system

### LINKED ISSUE #38: CoverImage management system

### PR DESCRIPTION
**Pull request by NosBos**
* Updated frontend to add image upload to listing form
* Updated backend APIs to support image upload
* Added tests

### CODE REVIEW

File: frontend/src/components/Seller/SellerListingForm.tsx [hunk 1]
```diff
@@ -117,6 +139,19 @@ export default function SellerListingForm(props: ISellerListingFormProps) {
...
         errors={errors}
       />
+      <Typography variant="h6"> Upload Cover Image </Typography>
+      {!isFileValidated && (
+        <Typography color={"red"}>
+          {" "}
+          Invalid File Type. Must be [png, jpg, jpeg]{" "}
+        </Typography>
+      )}
+      <input
```
Comment (Justgo13) [inline frontend/src/components/Seller/SellerListingForm.tsx]: could you wrap this like they do in the docs for button uniformity https://mui.com/material-ui/react-button/#upload-button

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/SecurityConfig.java [hunk 2]
```diff
@@ -25,7 +25,7 @@
         jsr250Enabled = true,
         prePostEnabled = true
 )
-public class SecurityConfig   {
```
Comment (kfoxIsProgrammer) [inline Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/SecurityConfig.java:28]: How dare you delete my space. JK

File: Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/SecurityConfig.java [hunk 3]
```diff
@@ -90,6 +91,8 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
                 .authorizeRequests()
                 .requestMatchers(HttpMethod.GET, "/listings")
                 .permitAll()
+                .requestMatchers(HttpMethod.GET, "/covers/*")
+                .permiteAll()
```
Comment (Justgo13) [inline Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/SecurityConfig.java]: permitAll()

Comment (kfoxIsProgrammer) [reply Backend/src/main/java/com/SYSC4806_Group13/SYSC4806_Project/Config/SecurityConfig.java]: Yes

**Review by Justgo13 — APPROVED** [review]

**Review by kfoxIsProgrammer — APPROVED** [review]

**Pull request merged**